### DISCUSSION TITLE
Installation without nvcc

### DISCUSSION THREAD

**Discussion by cromz22** [espnet/espnet#5767] [General]
Currently, ESPnet installation automatically detects the CUDA versions from the `nvcc` command.
However, in some environments, the `nvcc` command is not available or shows a different version from the one `nvidia-smi` shows. (cf. [Different CUDA versions shown by nvcc and NVIDIA-smi](https://stackoverflow.com/questions/53422407/different-cuda-versions-shown-by-nvcc-and-nvidia-smi))
In such cases, even though PyTorch can indeed work with GPUs, the Makefile installs a CPU version of PyTorch.

My current workaround is to finish the installation process as usual and then execute the following commands:
```
conda remove pytorch cpuonly
conda install pytorch pytorch-cuda=<version> -c pytorch -c nvidia
```
(The key is to remove the `cpuonly` package, which tells conda to install CPU version of pytorch. If you just reinstall pytorch conda will install CPU version again. The package exists only in conda and not in pip.)

This is not really an issue of ESPnet if CUDA environment is configured properly, so I'm just commenting this on a discussion. I hope this could be helpful to someone who stumbled into the same issue.

**Comment by sw005320** [comment]
Thanks, @cromz22, for TIPS!
